deployment/ccip/changeset/v1_5_1/cs_deploy_ping_pong.go

```diff
@@ -0,0 +1,198 @@
+package v1_5_1
+
+import (
+	"fmt"
+
+	"github.com/smartcontractkit/chainlink/deployment"
+	"github.com/smartcontractkit/chainlink/deployment/ccip/changeset"
+
+	"github.com/smartcontractkit/chainlink-ccip/chains/evm/gobindings/generated/latest/ping_pong_demo"
+)
+
+var _ deployment.ChangeSet[DeployPingPongDemoContractsConfig] = DeployPingPongDemoContractsChangeset
+var _ deployment.ChangeSet[StartPingPongDemoContractsConfig] = StartPingPongDemoContractsChangeset
+var _ deployment.ChangeSet[SetPausedPingPongDemoContractsConfig] = SetPausedPingPongDemoContractsChangeset
+var _ deployment.ChangeSet[SetConterpartPingPongDemoContractsConfig] = SetCounterpartPingPongDemoContractsChangeset
+
+type DeployPingPongDemoContractsConfig struct {
+	ChainSelector uint64
+	IsTestRouter  bool
+}
+
+func (c DeployPingPongDemoContractsConfig) Validate(env deployment.Environment, state changeset.CCIPOnChainState) error {
+	chainState := state.Chains[c.ChainSelector]
+
+	router := chainState.Router
+	if c.IsTestRouter {
+		router = chainState.TestRouter
+	}
+
+	if router == nil {
+		return fmt.Errorf("router address is empty for chain %d", c.ChainSelector)
+	}
+
+	if chainState.LinkToken == nil {
+		return fmt.Errorf("link token address is empty for chain %d", c.ChainSelector)
+	}
+
+	return nil
+}
+
+func DeployPingPongDemoContractsChangeset(env deployment.Environment, c DeployPingPongDemoContractsConfig) (deployment.ChangesetOutput, error) {
+	state, err := changeset.LoadOnchainState(env)
+	if err != nil {
+		return deployment.ChangesetOutput{}, fmt.Errorf("failed to load onchain state: %w", err)
+	}
+
+	if err := c.Validate(env, state); err != nil {
+		return deployment.ChangesetOutput{}, fmt.Errorf("invalid DeployPingPongDemoContractsConfig: %w", err)
+	}
+
+	newAB := deployment.NewMemoryAddressBook()
+
+	chain := env.Chains[c.ChainSelector]
+	chainState := state.Chains[c.ChainSelector]
+
+	router := chainState.Router
+	if c.IsTestRouter {
+		router = chainState.TestRouter
+	}
+
+	tv := deployment.NewTypeAndVersion(changeset.PingPongDemo, deployment.Version1_0_0)
+
+	addr, _, _, err := ping_pong_demo.DeployPingPongDemo(chain.DeployerKey, chain.Client, router.Address(), chainState.LinkToken.Address())
+	if err != nil {
+		return deployment.ChangesetOutput{}, fmt.Errorf("failed to deploy ping pong demo contract on %d: %w", c.ChainSelector, err)
+	}
+
+	err = newAB.Save(chain.Selector, addr.String(), tv)
+	if err != nil {
+		return deployment.ChangesetOutput{}, fmt.Errorf("failed to save ping pong demo contract address: %w", err)
+	}
+
```

Comment:
you need to use [DeployContract](https://github.com/smartcontractkit/chainlink/blob/develop/deployment/helpers.go#L164) here